packages/core/lib/events/DomainEventEmitter.ts

```diff
@@ -158,9 +147,10 @@ export class DomainEventEmitter<SupportedEvents extends CommonEventDefinition[]>
    * Register handler for all events supported by the emitter
    */
   public onAny(handler: AnyEventHandler<SupportedEvents>, isBackgroundHandler = false) {
-    for (const supportedEvent of this.eventRegistry.supportedEvents) {
-      this.on(supportedEvent.consumerSchema.shape.type.value, handler, isBackgroundHandler)
-    }
+    const eventTypes = this.eventRegistry.supportedEvents.map(
```

Comment:
probably we can calculate this value once during the instantiation? doesn't look like it ever changes

Reply:
Good point! will change it 🙏